src/Orleans.Core/Messaging/ClientMessageCenter.cs

```diff
@@ -137,16 +139,17 @@ internal class ClientMessageCenter : IMessageCenter, IDisposable
                         return gatewayConnections.Values.Count(conn => conn.IsLive);
                     }
                 });
-            if (StatisticsCollector.CollectQueueStats)
+            this.collectQueueStats = statisticsOptions.Value.CollectionLevel >= StatisticsLevel.Verbose2;
```

Comment:
this variable shouldn't be part of ClientMessagesCenter, and set value when ClientMessagesCenter was constructed. It should be with the StatisticGroup which collects QueueStats

Reply:
so maybe something like:
```
class QueueTrackingStatistic
{
public static bool ShouldCollect(CollectionLevel level) => level>= StatisticLevel.Verbose2;
public bool CollectQueueStats {get;set;}
```
And code here will be: `if(QueueTrackingStatistic.ShouldCollect(statisticOptions.CollectionLevel)) queueTracking = ...`

Reply:
logic on whether or not collect a set of statistics should stay with the StatisticGroup.